File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/BehandlingsnotatGetDto.java
package no.nav.melosys.tjenester.gui.dto;

import java.time.Instant;

import no.nav.melosys.domain.Behandlingsnotat;

public class BehandlingsnotatGetDto {

    private final long notatId; //
    private final String tekst; //
    private final Instant endretDato;
    private final Instant registrertDato;
    private final String behandlingstypeKode;
    private final boolean redigerbar; //
    private final String registrertAvNavn;

    public BehandlingsnotatGetDto(Behandlingsnotat behandlingsnotat, boolean redigerbar, String registrertAvNavn) {
        this.notatId = behandlingsnotat.getId();
        this.tekst = behandlingsnotat.getTekst();
        this.endretDato = behandlingsnotat.getEndretDato();
        this.registrertDato = behandlingsnotat.getRegistrertDato();
        this.behandlingstypeKode = behandlingsnotat.getBehandling().getType().getKode();

        this.redigerbar = redigerbar;
        this.registrertAvNavn = registrertAvNavn;
    }

    public long getNotatId() {
        return notatId;
    }

    public String getTekst() {
        return tekst;
    }

    public Instant getEndretDato() {
        return endretDato;
    }

    public Instant getRegistrertDato() {
        return registrertDato;
    }

    public String getBehandlingstypeKode() {
        return behandlingstypeKode;
    }

    public boolean isRedigerbar() {
        return redigerbar;
    }

    public String getRegistrertAvNavn() {
        return registrertAvNavn;
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/BehandlingsnotatGetDto.kt
package no.nav.melosys.tjenester.gui.dto

import no.nav.melosys.domain.Behandlingsnotat
import java.time.Instant

class BehandlingsnotatGetDto(
    behandlingsnotat: Behandlingsnotat, //
    val isRedigerbar: Boolean, val registrertAvNavn: String
) {
    val notatId: Long = behandlingsnotat.id //
    val tekst: String = behandlingsnotat.tekst //
    val endretDato: Instant = behandlingsnotat.endretDato
    val registrertDato: Instant = behandlingsnotat.registrertDato
    val behandlingstypeKode: String = behandlingsnotat.behandling.type.kode
    val behandlingstemaKode: String = behandlingsnotat.behandling.tema.kode
}
